src/store/base.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+import os
+from SPARQLWrapper import SPARQLWrapper, JSON, QueryResult
+
+class BaseStore(ABC):
+
+    # Every client runs the setup method
+    def __init__(self):
+        self.setup()
+
+    # Every client must have a setup method
+    @abstractmethod
+    def setup(self):
+        ...
+
+    @abstractmethod
+    def get_datasets(self):
+        ...
+
+
+class SparqlStore(BaseStore):
+
+    #seting up the self.url
+    def setup(self):
+        # set the url for sparql
+        self.url = os.getenv("SPARQL_ENDPOINT_URL")
+
+        #to make sure self.url is not none providin a default url
+        if self.url == None:
```

Comment:
can see where you're coming from but don't set a default. Just assert its not None (with a helpful message saying what the issue is).

the idea is you _want_ your apps to refuse start when they're not configured correctly - then loudly shout about precisely what the issue is. Defaults can be non obvious and obvious is your friend when it comes to maintaining things.